core/include/bitarithm.h

```diff
@@ -105,19 +110,44 @@ unsigned bitarithm_msb(unsigned v);
  *                  function will produce an infinite loop
  * @return          Bit Number
  *
- * Source: http://graphics.stanford.edu/~seander/bithacks.html#IntegerLogObvious
  */
-unsigned bitarithm_lsb(register unsigned v);
+static inline unsigned bitarithm_lsb(unsigned v);
 
 /**
  * @brief   Returns the number of bits set in a value
  * @param[in]   v   Input value
  * @return          Number of set bits
  *
- * Source: http://graphics.stanford.edu/~seander/bithacks.html#IntegerLogObvious
  */
 unsigned bitarithm_bits_set(unsigned v);
 
+/* implementations */
+
+static inline unsigned bitarithm_lsb(unsigned v)
+#if defined(BITARITHM_LSB_BUILTIN)
+{
+    return __builtin_ffs(v) - 1;
+}
+#elif defined(BITARITHM_LSB_LOOKUP)
+{
+/* Source: http://graphics.stanford.edu/~seander/bithacks.html#ZerosOnRightMultLookup */
+    extern const uint8_t MultiplyDeBruijnBitPosition[32];
+    return MultiplyDeBruijnBitPosition[((uint32_t)((v & -v) * 0x077CB531U)) >> 27];
```

Comment:
Maybe you could make the implementation *even faster* (TM), if you specialize the function to accept a `uint16_t` for the `SCHED_PRIO_LEVELS == 16` case, which is the default?

Reply:
The function is not used only by the scheduler. We could come up with an alternative (```lsb16()```), but I doubt on 32bit machines it would actually be significantly faster.

Please prove me wrong! ;)